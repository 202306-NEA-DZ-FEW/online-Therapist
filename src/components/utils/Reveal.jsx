import React, { useRef, useEffect } from "react";
import { motion, useInView, useAnimation } from "framer-motion";

export default function Reveal({ children }) {
    const ref = useRef(null);
    const isInView = useInView(ref, { once: true });

    const mainControls = useAnimation();

    useEffect(() => {
        if (isInView) {
            mainControls.start("visible");
        }
    }, [isInView]);
    return (
        <div ref={ref} style={{ position: "relative", overflow: "hidden" }}>
            <motion.div
                variants={{
                    hidden: { opacity: 0, y: 75 },
                    visible: { opacity: 1, y: 0 },
                }}
                initial='hidden'
                animate={mainControls}
                transition={{ duration: 0.5, delay: 0.25 }}
            >
                {children}
            </motion.div>
        </div>
    );
}
