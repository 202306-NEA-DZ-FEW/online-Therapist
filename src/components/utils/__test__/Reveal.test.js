import renderer from "react-test-renderer";
import Reveal from "../Reveal";

it("renders correctly", () => {
    const tree = renderer.create(<Reveal />).toJSON();
    expect(tree).toMatchSnapshot();
});
