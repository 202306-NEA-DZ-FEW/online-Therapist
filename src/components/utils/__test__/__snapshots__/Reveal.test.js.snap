// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(75px) translateZ(0)",
      }
    }
  />
</div>
`;
